docs/gitbook/getting_started/tutorial.md

```diff
@@ -0,0 +1,493 @@
+group by
+  t1.id
+;
+```
+
+Output for single sample can be:
+
+```
+id, probability
+ 1, 1.0261879540562902e-10
+```
+
+### Evaluation
+
+If you have test samples for evaluation, use Hivemall's [evaluation UDFs](http://hivemall.incubator.apache.org/userguide/eval/binary_classification_measures.html) to measure the accuracy of prediction.
+
+For instance, prediction accuracy over the `training` samples can be measured as:
+
+```sql
+with features_exploded as (
+  select
+    id,
+    extract_feature(fv) as feature,
+    extract_weight(fv) as value
+  from training t1 LATERAL VIEW explode(features) t2 as fv
+),
+predictions as (
+  select
+    t1.id,
+    sigmoid( sum(p1.weight * t1.value) ) as probability
+  from
+    features_exploded t1
+    LEFT OUTER JOIN classifier p1 ON (t1.feature = p1.feature)
+  group by
+    t1.id
+)
+select
+  auc(probability, label) as auc,
+  logloss(probability, label) as logloss
+from (
+  select t1.probability, t2.label
+  from predictions t1
+  join training t2 on (t1.id = t2.id)
+  ORDER BY probability DESC
+) t
+;
+```
+
+Since we are trying to solve the binary classification problem, the accuracy is measured by [Area Under the ROC Curve](https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve) [`auc()`](http://hivemall.incubator.apache.org/userguide/eval/auc.html) and/or [Logarithmic Loss](http://wiki.fast.ai/index.php/Log_Loss) [`logloss()`](http://hivemall.incubator.apache.org/userguide/eval/regression.html#logarithmic-loss).
+
+## Regression
+
+If you use [`train_regressor()`](http://hivemall.incubator.apache.org/userguide/misc/funcs.html#regression) instead of [`train_classifier()`](http://hivemall.incubator.apache.org/userguide/misc/funcs.html#binary-classification), you can also solve a regression problem with almost same queries.
+
+Imagine the following `customers` table:
+
+```sql
+create table if not exists customers
+(id bigint, gender String, age int, country string, num_purchases int)
+;
+```
+
+```sql
+insert overwrite table customers
+select 1 as id, "male" as gender, 23 as age, "Japan" as country, 12 as num_purchases
+union all
+select 2 as id, "female" as gender, 43 as age, "US" as country, 4 as num_purchases
+union all
+select 3 as id, "other" as gender, 19 as age, "UK" as country, 2 as num_purchases
+union all
+select 4 as id, "male" as gender, 31 as age, "US" as country, 20 as num_purchases
+union all
+select 5 as id, "female" as gender, 37 as age, "Australia" as country, 9 as num_purchases
+;
+```
+
+| gender | age | country | num_purchases |
+|:---:|:---|:---:|:---|
+| male | 23 |Japan | 12 |
+| female | 43 | US | 4 |
+| other | 19 | UK |  2 |
+| male | 31 | US |  20 |
+| female | 37 | Australia | 9 |
+
+Now, our goal is to build a regression model to predict the number of purchases potentially done by new customers.
+
+### Step 1. Feature representation
+
+Same as the classification example:
+
+```sql
+insert overwrite table training
```

Comment:
Same issue as classification. Let's drop INSERT and directly embed the query to the next training query:

```sql
create table regressor as
with training as (
  select
  id,
  array_concat(
    quantitative_features(
      array("age"),
      age
    ),
    categorical_features(
      array("country", "gender"),
      country, gender
    )
  ) as features,
  num_purchases
from
  customers
)
select
  train_regressor(
    features, -- feature vector
    num_purchases, -- target value
    '-loss_function squared -optimizer AdaGrad -regularization l2' -- hyper-parameters
  ) as (feature, weight)
from
  training
;
```

Reply:
Created training table. It's not affected.